sdk/src/androidTest/java/ly/count/android/sdk/UtilsTimeTests.java

```diff
@@ -49,10 +49,14 @@ public void testSeconds() {
     @Test
     public void testDiff() throws InterruptedException {
         long tms = UtilsTime.currentTimestampMs();
+        // TODO this test is flaky
+        // because current timestamp is generated by timeGenerator.uniqueTimestamp()
+        // which provides a unique timestamp every time it is called
         Thread.sleep(250);
 
         long tms2 = UtilsTime.currentTimestampMs();
 
+        // so it sometimes exceeds 260ms
```

Comment:
from the code of uniqueTimestamp it looks like it ensures last 10 timestamps are unique. With how big of a difference this fails?

Reply:
In some cases 330 in some 280 sometimes not

Reply:
so it differs above 10 ms which should not be the case. I think this is because cleanup was not correct at the top tests, they were clashing with this and was increasing it. So while it was sleeping, some other test became active and made the timer increase